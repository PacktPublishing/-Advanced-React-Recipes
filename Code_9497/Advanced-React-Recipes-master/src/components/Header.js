import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import PacktLogo from '../static/images/logo.png';

const HeaderLink = ({ children, ...props }) => (
  <NavLink
    exact
    className="p1 mx2 black rounded text-decoration-none"
    activeClassName="bg-white"
    {...props}
  >
    {children}
  </NavLink>
);

const Header = () => (
  <header className="flex justify-between items-center px4">
    <div>
      <h1 className="h1 inline-block mr2">🍽 MyRecipes</h1>
      <span className="mr1">by</span>
      <img src={PacktLogo} alt="by Packt" style={{ width: 20 }} />
    </div>
    <nav>
      <HeaderLink to="/">Home</HeaderLink>
      <HeaderLink to="/favorites">Favorites</HeaderLink>
    </nav>
  </header>
);

HeaderLink.propTypes = {
  children: PropTypes.node,
};

export default Header;
