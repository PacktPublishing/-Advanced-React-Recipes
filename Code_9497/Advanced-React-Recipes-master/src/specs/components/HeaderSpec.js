import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../../components/Header';

describe('<Header />', () => {
  let component;
  let tree;

  test('Should render Header', () => {
    component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
