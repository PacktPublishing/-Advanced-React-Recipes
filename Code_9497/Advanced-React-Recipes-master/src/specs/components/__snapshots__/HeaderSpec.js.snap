// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render Header 1`] = `
<header
  className="flex justify-between items-center px4"
>
  <div>
    <h1
      className="h1 inline-block mr2"
    >
      üçΩ MyRecipes
    </h1>
    <span
      className="mr1"
    >
      by
    </span>
    <img
      alt="by Packt"
      src="test-file-stub"
      style={
        Object {
          "width": 20,
        }
      }
    />
  </div>
  <nav>
    <a
      aria-current={false}
      className="p1 mx2 black rounded text-decoration-none"
      href="/"
      onClick={[Function]}
      style={undefined}
    >
      Home
    </a>
    <a
      aria-current={false}
      className="p1 mx2 black rounded text-decoration-none"
      href="/favorites"
      onClick={[Function]}
      style={undefined}
    >
      Favorites
    </a>
  </nav>
</header>
`;
